### DISCUSSION TITLE
Ability to Edit Taken Date

### DISCUSSION THREAD

**Discussion by Electro707** [LycheeOrg/Lychee#2537] [Ideas] [answered]
### Enhancement

It would be nice to edit the "Take Date", in the case of the photo not having that metadata for example

**Comment by ildyria** [comment]
I am not sure if we want to allow editing the displayed exif data.
There is the ability to edit the upload date though.
For the takedate, it is not much of a change, but I don't have time as I am working mostly on releasing v6.

**Reply by haivala** [reply]
This is something I need too because all pics do not have taken date and only way to order lots of albums in chronological order is to rename album with a date first.

**Reply by Electro707** [reply]
I can take a look at v5 over the weekend and see how trivial it would be for me to create the take date edit. Plus it would get me more familiar with the codebase (which is a bit confusion as somebody who mainly does C and Python programming).

I can also, after adding it in v5, see if I can also implement it in V6

**Reply by Electro707** [reply]
and to @haivala 's point, that is why I suggested specifically for the taken date, and sometimes in my case the upload date doesn't match the taken date.

**Reply by ildyria** [reply]
> I can take a look at v5

I would not suggest to touch the v5 given how close the v6 is. There will be no backward compatibility to v5 front-end, functions etc. v4 legacy interface will probably be still available though. But all the livewire stuff will be gone.

**Reply by ildyria** [reply]
> I can also, after adding it in v5, see if I can also implement it in V6

Go directly for the v6 route, skip v5.

Files to modify are:
- backend:
  - https://github.com/LycheeOrg/Lychee/blob/v6/app/Contracts/Http/Requests/RequestAttribute.php
  - https://github.com/LycheeOrg/Lychee/blob/v6/app/Http/Controllers/Gallery/PhotoController.php
  - https://github.com/LycheeOrg/Lychee/blob/v6/app/Http/Requests/Photo/EditPhotoRequest.php

- frontend:
  - https://github.com/LycheeOrg/Lychee/blob/v6/resources/js/components/drawers/PhotoEdit.vue
  - https://github.com/LycheeOrg/Lychee/blob/v6/resources/js/services/photo-service.ts

This should give you quite a head start.

**Reply by haivala** [reply]
I'm not sure how the ordering works in the background but is it enough that one pic has taken date to order the albums?
Another question is: should there be indication that taken date has been edited by lychee?

**Reply by ildyria** [reply]
> Another question is: should there be indication that taken date has been edited by lychee?

That is more of my wonder indeed. Should there be a flag that exif data have been tempered with. :)

**Reply by ildyria** [reply]
> I'm not sure how the ordering works in the background but is it enough that one pic has taken date to order the albums?

In theory yes.

**Comment by Puupuls** [comment]
This would make it a lot more usable as replacment to google photos as I frequently have to update multiple image created dates to be in correct timezone/time as camera photos are not always in sync with phone files

**Comment by ildyria** [comment, marked as answer]
https://github.com/LycheeOrg/Lychee/pull/2954
